### ISSUE TITLE
Put world editor content into `user://` and allow loading from there

### ISSUE THREAD

**Issue by HaSa1002** [Libre-TrainSim/Libre-TrainSim#191]
The data should be located in `user://tracks` or `user://editor`. Additionally direct loading of the project from that file location needs to be supported.

**Comment by Jean28518** [comment]
Mhmm...
The Track Editor content folder should be easily accessible for the user.
Maybe he want this folder to be on another harddrive.
Also this folder should be easily accessible from Libre TrainSim Resource Importer.

That's more than a few configuration files.

Won't change this...

**Comment by HaSa1002** [comment]
Using the `user://` file path gurantees us the directory to be writeable. Users don't need to access this directory directly. Instead we would put the resouce importer into this directory as well. Having the data stored there enables us to directly load content from the directory. We can put a "Open in File Explorer" button onto the track list. We need to save the path anyways...

**Comment by Jean28518** [comment]
Mhm, that's an argument for me that's okay!

**Comment by HaSa1002** [comment]
The directory structure is bad for git. You can't have the tracks as seperate git repositories, because every repo would need to be in the common parent.

Ilustration:
```
editor
 | .git (Hainfurt)
 | .git (U2-Nuremberg) <-- ouch.
 | .gitattributes (Hainfurt)
 | .gitattributes (U2)
 | Readme.md x2
 | LICENSE x2
 |--- Resources
 |     |--- Hainfurt
 |     |--- U2-Nuremberg
 \--- Worlds
       |--- Hainfurt
       |--- U2-Nuremberg
```

We need a structure which is comparable to:
```
editor
 \--- Worlds
       |--- Hainfurt
			|--- .git
			|--- Resources
			|--- World
			| Readme.md
			\ LICENSE
       |--- U2-Nuremberg
			|--- .git
			|--- Resources
			|--- World
			| Readme.md
			\ LICENSE
```

**Comment by Jean28518** [comment]
I see your point. What should we do with the basic resources?

**Comment by HaSa1002** [comment]
I would put them into the resource directory. We don't need to load worlds into the one folder. The structure is irrelevant execept for one place, where we look for maps.

**Comment by DasCapschen** [comment]
Hmm, do we want to load these under `Worlds`, or would it perhaps be better to use a more general directory name like `Mods`? And then everything could load like this:
```
editor
 \--- Mods
       |--- Hainfurt
			|--- .git/
			|--- Resources/
			|--- Trains/
			|--- Translations/
			|--- Worlds/
			| Readme.md
			\ LICENSE
```
In this case, everything can be loaded into a single folder, because if a mod would want to add a map and a train to go along with it...
```
editor
 \--- Trains
		|--- MyMod
			|--- .git/			<-- second git repo just for the train?
			\ MyTrain.tscn		<-- probably needs resources... from Worlds/MyMod ?
 \--- Worlds
       |--- MyMod
			|--- .git/			<-- separate git repo for the world
			|--- Resources/
			|--- World/
			| Readme.md
			\ LICENSE
 \--- Translations
		|--- MyMod
			|--- .git 			<-- a third git repo for translations?
			\ MyWorld.csv
```

**Comment by HaSa1002** [comment]
I would simply load them into their `Mods/<Mode Name>`. We don't need to require a dedicated file structure. The only thing that we do need is one resource `Mods/<Mode Name>/content.tres` that contains all the data that the game should now. Data like maps, trains, and probably scenarios as well. If wanted, a modder could extend from that ContentDescription resource and implement the _init() method. This method could do custom registration on the ContentLoader.

**Comment by HaSa1002** [comment]
TL;DR Just provide an easy way to register resources and don't care about the file system structure. Use the content loader to get data about content.

**Comment by DasCapschen** [comment]
I really like that idea :+1:

**Comment by Jean28518** [comment]
That's a great idea
